alf/environments/parallel_environment.py

```diff
@@ -141,6 +164,11 @@ def _reset(self):
         time_steps = [env.reset(self._blocking) for env in self._envs]
         if not self._blocking:
             time_steps = [promise() for promise in time_steps]
+        if self._spare_queue:
```

Comment:
resetting the spare envs can be done immediately after line 164.

Reply:
Good point.